### DISCUSSION TITLE
Help with getting Main.java to work

### DISCUSSION THREAD

**Discussion by dwpontius** [eclipse-theia/theia#13788] [Q&A] [answered]
Hello,

I am trying to refresh my memory about Java.  A few years ago I took some classes in programming and would like to see how much I remember. After a google search, I came across this IDE.

My problem is that I am trying to get a Main.java file working, and I can't figure out how to connect Java to it.  To save myself a bit of trouble, and despite it not being at all efficient, I tried using import on all of java (import java.*) and it says the package is not accessible.

I just want to go through [Java Programming](https://java-programming.mooc.fi/)'s content and write code to run on my own computer, and at the point I am at, I need to have a functioning Main.java file to proceed.

Any help would be appreciated.

P.S. - Searching for "package accessible" turned up a bunch of stuff that appears to not have anything to do with my problem.

**Comment by msujew** [comment]
Hey @dwpontius,

Theia IDE supports languages like Java via extensions. While we build the base IDE, those extension are usually written by other developers. In the case of Java, you're likely using the [redhat.java](https://github.com/redhat-developer/vscode-java) extension. You're probably better served asking there, since we don't have any developers here working on the Java integration.

Aside from that, I would personally recommend you to use a different IDE. Not because Theia isn't a good idea in general, but there are other, more beginner friendly IDEs for Java out there. Most people use [IntelliJ Community Edition](https://www.jetbrains.com/idea/download/), or the [Eclipse IDE](https://eclipseide.org/) which are both free and offer a deeper integration with Java than Theia currently offers.

**Reply by dwpontius** [reply, marked as answer]
Thanks for the response.

After a lot of trial and error, I managed to figure it out.  I had to make a new project, manually rename the class App to Main, and the file App.java to Main.java, and change the json file.  I'm not sure this was even necessary to get my custom class Person to work, but whatever.  Learning experience.
